### PR TITLE
Add support for Swift Package Manager

### PR DESCRIPTION
**Pull request by salling**
This PR addresses #965

Objective was to add support for swiftpm, while at the same time keeping the change set minimal. The "trick" here is to add a public header file that is used from Package.swift.

Note: this PR does not intend to expose YogaKit through SPM, only yoga itself.

### CODE REVIEW

File: yoga/swiftpm-support/umbrella.h [hunk 1]
```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+
+#ifdef __OBJC__
```
Comment (clausjoergensen) [inline yoga/swiftpm-support/umbrella.h:10]: Is it possible to use the [SWIFT_PACKAGE](https://github.com/apple/swift-package-manager/blob/main/Documentation/Usage.md#packaging-legacy-code) preprocessor flag here to make it even safer?

**Comment by clausjoergensen** [comment]
Ping #921

File: yoga/swiftpm-support/umbrella.h (hunk 1 above)
Comment (salling) [reply yoga/swiftpm-support/umbrella.h:10]: Does not seem to be defined. At least not in the context where this code is compiling.

**Comment by NickGerleman** [comment]
We are deprecating YogaKit as part of the Yoga 2.0 release. We are still going to release a new revision based on the current state of the repo, but won't be accepting new contributions, since we are going to be removing it from the repo after.

**Comment by clausjoergensen** [comment]
@NickGerleman this wasn't for YogaKit though. You keep updating the CocoaPods package, why can't you make a SPM package too?

File: Package.swift [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            publicHeadersPath: "./swiftpm-support",
+            cxxSettings: [
+                .headerSearchPath("../"),
+            ],
+            linkerSettings: [
+                .linkedFramework("Foundation"),
+            ]
+        )
+    ],
+    cxxLanguageStandard: .cxx14
```
Comment (cipolleschi) [inline Package.swift:34]: this should probably be `.cxx17`, as this is the version we are typically using internally.

File: Package.swift [hunk 3]
```diff
@@ -0,0 +1,35 @@
+// swift-tools-version:5.3
```
Comment (cipolleschi) [inline Package.swift:1]: we can probably use a more recent version of swift-tools here. At least `5.8`, until Xcode 15 become stable.

**Comment by cipolleschi** [comment]
Hi there, thanks @NickGerleman for tagging me.
I'm highly supportive of this change. Yoga is one of the few React Native packages that we can use independently and it makes a lot of sense to me to offer also a Swift PM alternative.

Also, the setup is fairly simple and I really like the folder with the public umbrella header.

I have not spent a lot of time thinking about how to validate it, though. One simple way could be to have an example app that import the dependency as a package and exercise it.
Another way could be to automatically generate the Package.swift from the podspec.

I don't have a strong opinion, although the second could be better in the long run. 
I'll leave the decision to you.

**Review by cipolleschi — COMMENTED** [review]
I'm very happy about this happening. It is a first step toward embracing more modern languages. I left a couple of comments to update a couple of values, but I'll leave you the last word on those.

File: Package.swift (hunk 2 above)
Comment (NickGerleman) [reply Package.swift:34]: Yoga is targeting C++14 for a little bit longer.

File: Package.swift (hunk 3 above)
Comment (NickGerleman) [reply Package.swift:1]: We should maybe consider bumping this in CocoaPods as well then, which is currently set on 5.1

**Comment by iainsmith** [comment]
This can be closed now we have https://github.com/facebook/yoga/commit/ef5784aca647a68343f5409e379519827edb86a9was

**Pull request closed without merging**